hopsworks-api/src/main/java/io/hops/hopsworks/api/airflow/AirflowService.java

```diff
@@ -0,0 +1,278 @@
+  }
+
+  @GET
+  @Path("purgeAirflowDagsLocal")
+  @Produces(MediaType.APPLICATION_JSON)
+  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+  public Response purgeAirflowDagsLocal(@Context HttpServletRequest req) {
+    Users user = userFacade.findByEmail(req.getRemoteUser());
+    String projectUsername = hdfsUsersController.getHdfsUserName(project, user);
+    airflowOperation(AirflowOp.PURGE_LOCAL, projectUsername);
+    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.OK).build();
+  }
+
+  @GET
+  @Path("restartWebserver")
+  @Produces(MediaType.APPLICATION_JSON)
+  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+  public Response restartAirflowWebserver(@Context HttpServletRequest req) {
+    Users user = userFacade.findByEmail(req.getRemoteUser());
+    String projectUsername = hdfsUsersController.getHdfsUserName(project, user);
+    airflowOperation(AirflowOp.RESTART_WEBSERVER, projectUsername);
+    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.OK).build();
+  }
+
+  @GET
+  @Path("copyFromAirflowToHdfs")
+  @Produces(MediaType.APPLICATION_JSON)
+  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+  public Response copyFromAirflowToHdfs(@Context HttpServletRequest req) {
+    Users user = userFacade.findByEmail(req.getRemoteUser());
+    String projectUsername = hdfsUsersController.getHdfsUserName(project, user);
+    airflowOperation(AirflowOp.TO_HDFS, projectUsername);
+    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.OK).build();
+  }
+
+  @GET
+  @Path("copyToAirflowFromHdfs")
+  @Produces(MediaType.APPLICATION_JSON)
+  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+  public Response copyToAirflowFromHdfs(@Context HttpServletRequest req) {
+    Users user = userFacade.findByEmail(req.getRemoteUser());
+    String projectUsername = hdfsUsersController.getHdfsUserName(project, user);
+    airflowOperation(AirflowOp.FROM_HDFS, projectUsername);
+    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.OK).build();
+
+  }
+
+  public boolean airflowOperation(AirflowOp op, String projectUsername) {
+
+    try {
+      Logger.getLogger(AirflowService.class.getName()).log(Level.INFO, "Attempting to restart airflow webserver");
+      String script = settings.getHopsworksDomainDir() + "/bin/airflowOps.sh";
+      String copyCommand = op.toString();
+
+      String[] command;
+      if (op == op.RESTART_WEBSERVER) {
+        String[] c = {"sudo", script, copyCommand, project.getName(), projectUsername};
+        command = c;
+      } else {
+        String[] c = {script, copyCommand, project.getName(), projectUsername};
+        command = c;
+      }
+      Logger.getLogger(AirflowService.class.getName()).log(Level.INFO, "Command: {0}", command);
+      ProcessBuilder ps = new ProcessBuilder(command);
```

Comment:
Here the `OSProcessExecutor` should be used. The way this method is written might lead to resources not being freed up correctly and problems with glassfish.
As bonus point, put the log level to `DEBUG`